### PR TITLE
Feat/filter by genre

### PR DESCRIPTION
**Pull request by rinaehyun**

### CODE REVIEW

File: frontend/src/components/filterPage/FilterPage.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+type FilterPageProps = {
+    selectedGenre: string,
+    setSelectedGenre: (genre: string) => void,
+    handleApplyFilter: (genre: string) => void,
+    setShowKeywordTag: Dispatch<SetStateAction<boolean>>
+}
+
+export default function FilterPage({ setSelectedGenre, handleApplyFilter, setShowKeywordTag }: FilterPageProps) {
+    const [isDisable, setIsDisable] = useState<boolean>(true);
```
Comment (esgoet) [inline frontend/src/components/filterPage/FilterPage.tsx:13]:
```suggestion
    const [isDisabled, setIsDisabled] = useState<boolean>(true);
```

File: frontend/src/components/filterPage/FilterPage.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    const [genre, setGenre] = useState<string>('Select');
+
+    const genres : string[] = ["NONE", "FICTION", "MYSTERY", "THRILLER",
```
Comment (esgoet) [inline frontend/src/components/filterPage/FilterPage.tsx:16]:
```suggestion
    const genres : Genre[] = ["NONE", "FICTION", "MYSTERY", "THRILLER",
```
created a new Type Genre that we can use for this

File: frontend/src/pages/BookGalleryPage/bookGalleryPage/BookGalleryPage.tsx [hunk 3]
```diff
@@ -16,13 +17,59 @@ export default function BookGalleryPage({filteredBooks, setSearchInput}: BookGal
     const [statusFilter, setStatusFilter] = useState<string>("ALL")
 
 
+
+    const [showFilter, setShowFilter] = useState<boolean>(false);
+    const [selectedGenre, setSelectedGenre] = useState<string>("Select");
+    const [showFilterTag, setShowFilterTag] = useState<boolean>(false);
+
+    const handleClick = () => {
```
Comment (esgoet) [inline frontend/src/pages/BookGalleryPage/bookGalleryPage/BookGalleryPage.tsx:25]: handleClick is unnecessary, could just have onClick={()=>setShowFilter(!showFilter))? but if you want to keep it, I'd rather the function has a more meaningful name, like toggleFilter

File: frontend/src/pages/BookGalleryPage/bookGalleryPage/BookGalleryPage.tsx [hunk 4]
```diff
@@ -16,13 +17,59 @@ export default function BookGalleryPage({filteredBooks, setSearchInput}: BookGal
...
+                        handleApplyFilter={handleApplyFilter}
+                        setShowKeywordTag={setShowFilterTag}
+                    />
+                }
+                {(showFilterTag && !showFilter) &&
+                    <div className={"filter-tag-area"}>
+                        <p style={{fontStyle: "italic", fontWeight: "bold"}}>Filtered Genre: </p>
+                        <div className={"filter-tag"}>
+                            <p className={"filter-tag-genre"}>{formatEnum(selectedGenre)}</p>
+                            <button className={"filter-tag-close"} onClick={handleRemoveFilter}>x</button>
```
Comment (esgoet) [inline frontend/src/pages/BookGalleryPage/bookGalleryPage/BookGalleryPage.tsx:63]: would make the filter name part of the button, or make it clearer, that only the x is the button
```suggestion
                            <button className={"filter-tag-close"} onClick={handleRemoveFilter}>{formatEnum(selectedGenre)} x</button>
```

File: frontend/src/components/filterPage/FilterPage.tsx (hunk 1 above)
Comment (rinaehyun) [reply frontend/src/components/filterPage/FilterPage.tsx]: erledgit

File: frontend/src/components/filterPage/FilterPage.tsx (hunk 2 above)
Comment (rinaehyun) [reply frontend/src/components/filterPage/FilterPage.tsx]: erledgit

File: frontend/src/pages/BookGalleryPage/bookGalleryPage/BookGalleryPage.tsx (hunk 3 above)
Comment (rinaehyun) [reply frontend/src/pages/BookGalleryPage/bookGalleryPage/BookGalleryPage.tsx]: erledgit

File: frontend/src/pages/BookGalleryPage/bookGalleryPage/BookGalleryPage.tsx (hunk 4 above)
Comment (rinaehyun) [reply frontend/src/pages/BookGalleryPage/bookGalleryPage/BookGalleryPage.tsx]: erledgit

**Pull request merged**